python-frontend/src/main/java/org/sonar/python/semantic/ProjectLevelSymbolTable.java

```diff
@@ -206,20 +210,35 @@ private void addModuleV2(FileInput astRoot, String packageName, PythonFile pytho
   }
 
   private class DjangoViewsVisitor extends BaseTreeVisitor {
+
+    String fullyQualifiedModuleName;
+
+    public DjangoViewsVisitor(String fullyQualifiedModuleName) {
+      this.fullyQualifiedModuleName = fullyQualifiedModuleName;
+    }
+
     @Override
     public void visitCallExpression(CallExpression callExpression) {
-      Symbol calleeSymbol = callExpression.calleeSymbol();
-      if (calleeSymbol == null) {
-        return;
-      }
-      if ("django.urls.conf.path".equals(calleeSymbol.fullyQualifiedName())) {
+      super.visitCallExpression(callExpression);
+      if (isCallRegisteringDjangoView(callExpression)) {
         RegularArgument viewArgument = nthArgumentOrKeyword(1, "view", callExpression.arguments());
         if (viewArgument != null) {
-          getSymbolFromTree(viewArgument.expression())
-            .filter(symbol -> symbol.fullyQualifiedName() != null)
-            .ifPresent(symbol -> djangoViewsFQN.add(symbol.fullyQualifiedName()));
+          PythonType pythonType = viewArgument.expression().typeV2();
+          if (pythonType instanceof UnknownType.UnresolvedImportType unresolvedImportType) {
```

Comment:
Since we will need to add FQN for other types (at least for `ClassType`) I suggest to introduce an interface like `HasFullyQualifiedName`, and implement it for the `UnresolvedImportType` and for the `FunctionType`

Reply:
As discussed, `UnresolvedImportType` does not have a fully qualified name, because we said fully qualified name is strictly the name where a type is defined, and not just a re-exported name. Because the type is unknown, it has an `importPath` instead, which I think is good to keep as a clearly distinct concept from `fullyQualifiedName`.

In this case, I think what is misleading is probably the fact that I kept `djangoViewsFqns` when it probably should be `djangoViewsImportPaths` (`fullyQualifiedName` being seen here as a more precise import path).

Wdyt?

If we agree on that, I can rename the variable for `djangoViewsFqns`. As for the `HasFullyQualifiedName` interface, I agree it will make sense when introduced for `ClassType`, though. I'd probably introduce it at the same time as `ClassType` however.

Reply:
As discussed, `UnresolvedImportType` does not have a fully qualified name, because we said fully qualified name is strictly the name where a type is defined, and not just a re-exported name. Because the type is unknown, it has an `importPath` instead, which I think is good to keep as a clearly distinct concept from `fullyQualifiedName`.

In this case, I think what is misleading is probably the fact that I kept `djangoViewsFqns` when it probably should be `djangoViewsImportPaths` (`fullyQualifiedName` being seen here as a more precise import path).

Wdyt?

If we agree on that, I can rename the variable for `djangoViewsFqns`. As for the `HasFullyQualifiedName` interface, I agree it will make sense when introduced for `ClassType`, though. I'd probably introduce it at the same time as `ClassType` however.